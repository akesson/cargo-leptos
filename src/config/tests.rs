use camino::Utf8PathBuf;

use super::Config;

fn opts(project: Option<&str>) -> crate::Opts {
    crate::Opts {
        release: false,
        project: project.map(|s| s.to_string()),
        verbose: 0,
        bin_features: Vec::new(),
        lib_features: Vec::new(),
    }
}

#[test]
fn test_project() {
    let cli = opts(None);

    let conf = Config::load(cli, &Utf8PathBuf::from("examples/project/Cargo.toml"), true).unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace() {
    let cli = opts(None);

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project1() {
    let cli = opts(Some("project1"));

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project2() {
    let cli = opts(Some("project2"));

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}
