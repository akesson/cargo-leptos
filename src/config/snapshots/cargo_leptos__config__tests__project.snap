---
source: src/config/tests.rs
expression: conf
---
Config {
    projects: [
        Project {
            name: "example",
            lib: LibPackage {
                name: "example",
                dir: ".",
                wasm_file: SourcedSiteFile {
                    source: "target/front/wasm32-unknown-unknown/debug/example.wasm",
                    dest: "target/site/pkg/example.wasm",
                    site: "pkg/example.wasm",
                },
                js_file: SiteFile {
                    dest: "target/site/pkg/example.js",
                    site: "pkg/example.js",
                },
                features: [
                    "hydrate",
                ],
                default_features: false,
                output_name: "example",
                path_deps: "",
            },
            bin: BinPackage {
                name: "example",
                dir: ".",
                exe_file: "target/server/debug/example",
                target: "example",
                features: [
                    "ssr",
                ],
                default_features: false,
                path_deps: "",
            },
            style: Some(
                StyleConfig {
                    file: SourcedSiteFile {
                        source: "style/main.scss",
                        dest: "target/site/pkg/example.css",
                        site: "pkg/example.css",
                    },
                    browserquery: "defaults",
                },
            ),
            watch: true,
            release: false,
            site: Site {
                addr: 127.0.0.1:3000,
                reload: 127.0.0.1:3001,
                root_dir: "target/site",
                pkg_dir: "pkg",
                file_reg: {},
                ext_file_reg: {},
            },
            end2end: Some(
                 {
                    cmd: "npx playwright test",
                    dir: "",
                },
            ),
            assets: Some(
                AssetsConfig {
                    dir: "assets",
                },
            ),
        },
    ],
    cli: Opts {
        release: false,
        project: None,
        lib_features: [],
        bin_features: [],
        verbose: 0,
    },
    watch: true,
}
